src/main/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStep.java

```diff
@@ -140,15 +143,16 @@
                 b.append(Pattern.quote(secret));
             }
             pattern = Secret.fromString(b.toString());
+            this.charsetName = charsetName;
         }
 
         @Override public OutputStream decorateLogger(AbstractBuild _ignore, final OutputStream logger) throws IOException, InterruptedException {
             final Pattern p = Pattern.compile(pattern.getPlainText());
             return new LineTransformationOutputStream() {
                 @Override protected void eol(byte[] b, int len) throws IOException {
-                    Matcher m = p.matcher(new String(b, 0, len));
+                    Matcher m = p.matcher(new String(b, 0, len, Charset.forName(charsetName)));
```

Comment:
I have a fix for this. But before pushing it I wanted to test it. I have not been able to reproduce the issue at all with the plugin as it is in this PR. Had a job using `withCredentials` that sleeps for 100 seconds using latest version of Credentials Binding. While the job is running I upgrade the plugin manually to the one in this PR, restart Jenkins instance and all is good. Job finishes successfully but it should not, right? Did I misunderstand something?